packages/replay/src/replay.ts

```diff
@@ -921,8 +921,9 @@ export class ReplayContainer implements ReplayContainerInterface {
     const client = hub.getClient();
     const scope = hub.getScope();
     const transport = client && client.getTransport();
+    const dsn = client?.getDsn();
 
-    if (!client || !scope || !transport) {
+    if (!client || !scope || !transport || !dsn) {
```

Comment:
just adding this here so we can be certain, `dsn` isn't `undefined` further below. In case we don't have a dsn, we can't send anything anyway.